### PR TITLE
OTF2 Reader: Performance Improvements

### PR DESCRIPTION
**Pull request by adityaranjan**
- gives user option to choose how many cores to parallelize with
- uses floor(0.75 * num_cores) as default for parallelization
- concatenates dataframes instead of dictionaries

### CODE REVIEW

File: pipit/readers/otf2_reader.py [hunk 1]
```diff
@@ -13,10 +13,16 @@
 class OTF2Reader:
     """Reader for OTF2 trace files"""
 
-    def __init__(self, dir_name):
+    def __init__(self, dir_name, num_parallel=None):
```
Comment (bhatele) [inline pipit/readers/otf2_reader.py]: Rename num_parallel to num_processes

File: pipit/readers/otf2_reader.py [hunk 2]
```diff
@@ -231,14 +237,16 @@ def events_reader(self, rank_size):
...
+        return pd.DataFrame(
+            {
+                "Timestamp (ns)": timestamps,
+                "Event Type": event_types,
+                "Name": names,
+                "Thread": thread_ids,
+                "Process": process_ids,
+                "Attributes": event_attributes,
+            }
+        )
```
Comment (bhatele) [inline pipit/readers/otf2_reader.py:249]: why this change? Also the comment is now outdated

File: pipit/trace.py [hunk 3]
```diff
@@ -17,12 +17,12 @@ def __init__(self, definitions, events):
         self.events = events
 
     @staticmethod
-    def from_otf2(dirname):
+    def from_otf2(dirname, num_parallel=None):
```
Comment (bhatele) [inline pipit/trace.py]: Rename num_parallel to num_processes

**Review by bhatele — CHANGES_REQUESTED** [review]
See inline

File: pipit/readers/otf2_reader.py (hunk 2 above)
Comment (adityaranjan) [reply pipit/readers/otf2_reader.py:249]: following what alex does, this removes need for the previous combining of dictionaries, which was done confusingly and was ambiguous
